pinot-query-planner/src/main/java/org/apache/calcite/sql/fun/PinotOperatorTable.java

```diff
@@ -115,25 +109,207 @@ public final void initNoDuplicate() {
     //   2. register special handling that differs from calcite standard.
     for (TransformFunctionType transformFunctionType : TransformFunctionType.values()) {
       if (transformFunctionType.getSqlKind() != null) {
+        Set<String> registeredFunctionNames = new HashSet<>();
+        String transformFunctionTypeName = transformFunctionType.getName();
         // 1. Register the aggregation function with Calcite
-        registerTransformFunction(transformFunctionType.getName(), transformFunctionType);
+        registerTransformFunction(transformFunctionTypeName, transformFunctionType);
+        registeredFunctionNames.add(transformFunctionTypeName.toUpperCase());
+
         // 2. Register the aggregation function with Calcite on all alternative names
         List<String> alternativeFunctionNames = transformFunctionType.getAlternativeNames();
         for (String alternativeFunctionName : alternativeFunctionNames) {
+          if (registeredFunctionNames.contains(alternativeFunctionName.toUpperCase())) {
+            continue;
+          }
           registerTransformFunction(alternativeFunctionName, transformFunctionType);
+          registeredFunctionNames.add(alternativeFunctionName.toUpperCase());
         }
       }
     }
   }
 
+  private void initPinotOverrideSql() {
+    register(new PinotSqlCoalesceFunction());
+  }
+
+  private void initCalciteStandardSql() {
```

Comment:
Good idea.

Reply:
Is it possible to also see if we can somehow sync with the parser.jj config? We have a sunset of Babel conformance keywords registered there. I doubt we need that many.